storm-core/src/jvm/org/apache/storm/daemon/supervisor/timer/RunProfilerActions.java

```diff
@@ -0,0 +1,221 @@
+                            boolean stop = System.currentTimeMillis() > profileRequest.get_time_stamp() ? true : false;
+                            Long port = profileRequest.get_nodeInfo().get_port().iterator().next();
+                            String targetDir = ConfigUtils.workerArtifactsRoot(conf, String.valueOf(port));
+                            Map stormConf = ConfigUtils.readSupervisorStormConf(conf, stormId);
+
+                            String user = null;
+                            if (stormConf.get(Config.TOPOLOGY_SUBMITTER_USER) != null) {
+                                user = (String) (stormConf.get(Config.TOPOLOGY_SUBMITTER_USER));
+                            }
+                            Map<String, String> env = null;
+                            if (stormConf.get(Config.TOPOLOGY_ENVIRONMENT) != null) {
+                                env = (Map<String, String>) stormConf.get(Config.TOPOLOGY_ENVIRONMENT);
+                            } else {
+                                env = new HashMap<String, String>();
+                            }
+
+                            String str = ConfigUtils.workerArtifactsPidPath(conf, stormId, port.intValue());
+                            StringBuilder stringBuilder = new StringBuilder();
+                            FileReader reader = null;
+                            BufferedReader br = null;
+                            try {
+                                reader = new FileReader(str);
+                                br = new BufferedReader(reader);
+                                int c;
+                                while ((c = br.read()) >= 0) {
+                                    stringBuilder.append(c);
+                                }
+                            } catch (IOException e) {
+                                if (reader != null)
+                                    reader.close();
+                                if (br != null)
+                                    br.close();
+                            }
+                            String workerPid = stringBuilder.toString().trim();
+                            ProfileAction profileAction = profileRequest.get_action();
+                            String logPrefix = "ProfilerAction process " + stormId + ":" + port + " PROFILER_ACTION: " + profileAction + " ";
+
+                            // Until PROFILER_STOP action is invalid, keep launching profiler start in case worker restarted
+                            // The profiler plugin script validates if JVM is recording before starting another recording.
+                            String command = mkCommand(profileAction, stop, workerPid, targetDir);
+                            List<String> listCommand = new ArrayList<>();
+                            if (command != null) {
+                                listCommand.addAll(Arrays.asList(command.split(" ")));
+                            }
```

Comment:
Instead of returning a string and converting it to a List, can we just return the list, that way we don't have to worry about white space anywhere in the name of the profile command etc.